MWI: Add new JWT CA for signing bound keypair documents

timothyb89 opened the pull request:
This adds a new JWT CA for issuing JWT documents related to bound keypair joining, particularly client state document that clients are expected to present back to auth when rejoining.

timothyb89 committed:
MWI: Add new JWT CA for signing bound keypair documents

This adds a new JWT CA for signing JWT documents related to bound
keypair joining, particularly client state document that clients are
expected to present back to auth when rejoining.

greedy52 commented on lib/cryptosuites/suites.go:
```diff
@@ -202,6 +205,7 @@ var (
 		GitClient:           Ed25519,
 		AWSRACATLS:          ECDSAP256,
 		BoundKeypairJoining: Ed25519,
+		BoundKeypairCAJWT:   Ed25519,
```
New CAs should use `ECDSAP256` for legacy to be compatible with KMS/HSM.

https://github.com/gravitational/teleport/blob/master/rfd/0136-modern-signature-algorithms.md#legacy-suite

I learned it the hard way and had to rollback https://github.com/gravitational/teleport/pull/53444

timothyb89 replied:
Yikes, good catch! I've swapped the algorithm like you suggested.

timothyb89 replied:
...and I've also added a note about this with a link to the RFD to hopefully avoid this in the future.

greedy52 commented on tool/tctl/common/auth_command.go:
```diff
@@ -221,6 +221,7 @@ var allowedCertificateTypes = []string{
 	"saml-idp",
 	"github",
 	"awsra",
+	"bound_keypair",
```
should this be consistent with other types above using `-` instead of `_`?

what's the format being exported? seem current only TLS and SSH keys are exported in ca_export.go.

timothyb89 replied:
Huh, I figured this list was used for `tctl auth rotate`, but apparently it only applies to `export`. We don't seem to export JWT CAs so I'll just remove it from the list entirely.

timothyb89 committed:
Fix issues from code review

Removes `bound_keypair` from the list of exportable CAs in tctl, and
switches the `legacy` suite algorithm to ECDSAP256 due to hsm
compatibility requirements.

timothyb89 committed:
Add note about FIPS/HSM compatibility for CA suites
